Implements merge operator `+>` (#502)
arr-ai/arrai: docs/example.md

$ arrai eval '{|a,b,c| (1,2,2), (1,2,3)} <&> {|b,c,d| (2,3,4), (1,3,4)}'
{(a: 1, b: 2, c: 3, d: 4)}
```

> ### Merging operator

Reviewer: Nit: For consistency with above operators.

```suggestion
### Merge
```

> Merges two tuples and combines all rhs(right-hand side) attributes and any lhs(left-hand side) attributes whose name isn't on the rhs into a new tuple.

Reviewer: Small nit: good description but a bit of a mouthful. Two sentences is clearer, saying a) what it does, and b) addressing edge cases.

```suggestion
Merge combines two tuples, producing a single tuple containing a union of their attributes. If the same name is present in both the LHS (left-hand side) and RHS (right-hand side) tuples, the RHS value takes precedence in the output.
```

#### Examples

```bash

> arrai e '(a: 1, b: 2) +> (b: 3, c: 4)'

Reviewer: ```suggestion
$ arrai e '(a: 1, b: 2) +> (b: 3, c: 4)'
```

(a: 1, b: 3, c: 4)
```

```bash

> arrai e '(a: 1, b: (c: 2)) +> (b: (c: 4), c: 4)'

Reviewer: ```suggestion
$ arrai e '(a: 1, b: (c: 2)) +> (b: (c: 4), c: 4)'
```

> (a: 1, b: (c: 4), c: 4)

Reviewer: This is a nice example, but it leaves it a little ambiguous as to whether the merge is "deep" (i.e. recursive) or "shallow" (i.e. just replace the whole attribute value).

For example:

```
$ arrai e '(a: (b: 1)) +> (a: (c: 2))'
```

Would this produce `(a: (b: 1, c: 2))` or `(a: (c: 2))`? I'm guessing the latter but ideally the docs would tell me so I don't have to guess.
